tedge/src/system_services/error.rs

```diff
@@ -36,3 +39,16 @@ pub enum SystemdError {
     #[error("Returned exit code: '{code:?}' for: systemd' is unhandled.")]
     UnhandledReturnCode { code: i32 },
 }
+
+/// The error type used by the `OpenRcServiceManager`
+#[derive(thiserror::Error, Debug)]
+pub enum OpenRcServiceError {
+    #[error("OpenRC is not available on the system.")]
+    ServiceManagerNotAvailable,
+
+    #[error("Service command <{service_command:?}> failed with code: {code:?}.")]
+    ServiceCommandFailed {
+        service_command: String,
+        code: Option<i32>,
+    },
+}
```

Comment:
Do we need an error type per init system?

Looking at this one, it seems that is can be generic provided a service name.